.grit/patterns/codecept_to_playwright.md

```diff
@@ -12,32 +12,52 @@ language js
 
 pattern convert_base_page() {
     `export default { $properties }` where {
-        $program <: contains `const { I } = inject();`,
+        $program <: contains `const { I } = inject();` => .,
         $properties <: maybe contains bubble or {
             pair($key, $value) as $pair where {
                 $pair => `get $key() { return $value }`
             },
             `locate($locator).as($_)` => `this.page.locator($locator)`,
+            method_definition($async, $static) as $method where {
+                $async <: false,
+                $method => `async $method`,
+            },
+            `I.waitInUrl($url)` => `await this.page.waitForUrl(new RegExp($url))`,
+            `I.waitForLoader()` => `await this.waitForLoader()`,
+            `I.waitForText($text, $timeout, $target)` => `await expect($target).toHaveText($text, {
+                timeout: $timeout * 1000,
+                ignoreCase: true,
+            })`,
+            `I.see($text, $target)` => `await expect($target).toContainText($text)`,
+            `I.waitForVisible($target)` => `await $target.waitFor({ state: 'visible' })`,
+            `I.click($target)` => `await $target.click()`,
         },
-        $filename <: r".*/?([^/]+)\.[a-zA-Z]*"($base_name),
+        $filename <: r".*?/?([^/]+)\.[a-zA-Z]*"($base_name),
     } => `export default class $base_name extends BasePage {
         $properties
     }`
 }
 
 pattern remove_commas() {
-    r"(get\s+\w+\s*\(\s*\)\s*\{[^}]*\})\s*,"($getter) => $getter,
+    or {
+        r"(?s)(get\s+\w+\s*\(\s*\)\s*\{[^}]*\})\s*,"($getter) => $getter,
+        // Hack to remove the incorrect trailing comma
+        `async $method($params) { $body }` => `async $method($params) {
+    $body
+}`,
+    }
 }
```

Comment:
I think we probably want to move this to a built-in cleanup? A trailing comma is never valid in this context, so I don't see a harm in removing the commas in our post-processing pipeline.

Reply:
I don't think we should make these, especially the apparent identity rewrite, built ins until/unless we get commas working in their own right in snippets. The async identity rewrite in particular is going to add some amount of performance overhead at the moment since it overmatches a lot of code.

Reply:
Sure, we don't have to do it immediately. I agree the identity overwrite is less obvious.